lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -509,84 +514,96 @@ defmodule ExUnit.DocTest do
   end
 
   # Expression numbers are simply skipped.
-  defp extract_tests([{<<"iex(", _>> <> string, line} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests([{"iex" <> skip_iex_number(string), line} | lines], expr_acc, expected_acc, acc, newtest)
+  defp extract_tests([{<<"iex(", _>> <> string = line, line_no} | lines],
+                     expr_acc, expected_acc, acc, new_test, module) do
+    extract_tests([{"iex" <> skip_iex_number(string, module, line_no, line), line_no} | lines],
+                     expr_acc, expected_acc, acc, new_test, module)
   end
 
   # Expression numbers are simply skipped redux.
-  defp extract_tests([{<<"...(", _>> <> string, line} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests([{"..." <> skip_iex_number(string), line} | lines], expr_acc, expected_acc, acc, newtest)
+  defp extract_tests([{<<"...(", _>> <> string, line_no} = line | lines],
+                     expr_acc, expected_acc, acc, new_test, module) do
+    extract_tests([{"..." <> skip_iex_number(string, module, line_no, line), line_no} | lines],
+                  expr_acc, expected_acc, acc, new_test, module)
   end
 
   # Skip empty or documentation line.
-  defp extract_tests([_ | lines], "", "", acc, _) do
-    extract_tests(lines, "", "", acc, true)
+  defp extract_tests([_ | lines], "", "", acc, _, module) do
+    extract_tests(lines, "", "", acc, true, module)
   end
 
   # Encountered an empty line, store pending test
-  defp extract_tests([{"", _} | lines], expr_acc, expected_acc, [test | t], _) do
+  defp extract_tests([{"", _} | lines], expr_acc, expected_acc, [test | acc_rest], _, module) do
     test = add_expr(test, expr_acc, expected_acc)
-    extract_tests(lines, "", "", [test | t], true)
+    extract_tests(lines, "", "", [test | acc_rest], true, module)
   end
 
   # Finally, parse expected_acc.
-  defp extract_tests([{expected, _} | lines], expr_acc, "", acc, newtest) do
-    extract_tests(lines, expr_acc, expected, acc, newtest)
+  defp extract_tests([{expected, _} | lines], expr_acc, "", acc, new_test, module) do
+    extract_tests(lines, expr_acc, expected, acc, new_test, module)
   end
 
-  defp extract_tests([{expected, _} | lines], expr_acc, expected_acc, acc, newtest) do
-    extract_tests(lines, expr_acc, expected_acc <> "\n" <> expected, acc, newtest)
+  defp extract_tests([{expected, _} | lines], expr_acc, expected_acc, acc, new_test, module) do
+    extract_tests(lines, expr_acc, expected_acc <> "\n" <> expected, acc, new_test, module)
   end
 
-  defp skip_iex_number(")>" <> string) do
+  defp skip_iex_number(")>" <> string, _module, _line_no, _line) do
     ">" <> string
   end
 
-  defp skip_iex_number(<<_>> <> string) do
-    skip_iex_number(string)
+  defp skip_iex_number("", module, line_no, line) do
+    raise Error, line: line_no, module: module,
+                 message: ~s{unknown IEx prompt: #{inspect line}.\n} <>
```

Comment:
I think we can define the message in a previous line and then call error:

```elixir
message = "unknown IEx prompt: #{inspect line}. Accepted formats are: iex>, iex(1)>, ...>, ...(1)>"
raise Error, line: line_no, module: module, message: message
```

I think it will be clear than nesting as today.